common/services/deviceOperation.js

```diff
@@ -1,224 +1,221 @@
-  var sPayload = doordeck.libSodium.to_base64(JSON.stringify(oPayload), doordeck.libSodium.base64_variants.ORIGINAL)
-  var message = sHeader + '.' + sPayload
-  var signature = doordeck.libSodium.crypto_sign_detached(message, privateKey)
-  return message + '.' + doordeck.libSodium.to_base64(signature, doordeck.libSodium.base64_variants.ORIGINAL)
-}
+  // noinspection JSUnresolvedVariable,JSUnresolvedFunction
+  const sHeader = doordeckSDK.libSodium.to_base64(JSON.stringify(oHeader), doordeckSDK.libSodium.base64_variants.ORIGINAL)
+  // noinspection JSUnresolvedVariable,JSUnresolvedFunction
+  const sPayload = doordeckSDK.libSodium.to_base64(JSON.stringify(oPayload), doordeckSDK.libSodium.base64_variants.ORIGINAL)
+  const message = sHeader + '.' + sPayload
+  // noinspection JSUnresolvedVariable,JSUnresolvedFunction
+  const signature = doordeckSDK.libSodium.crypto_sign_detached(message, privateKey)
+  // noinspection JSUnresolvedVariable,JSUnresolvedFunction
+  return message + '.' + doordeckSDK.libSodium.to_base64(signature, doordeckSDK.libSodium.base64_variants.ORIGINAL)
+};
 
-var executor = function (deviceId, operation) {
-  var signature = signer(deviceId, operation)
-  return axios.post(baseUrl + '/device/' + deviceId + '/execute', signature,
-    {
-      skipAuthorization: true,
-      headers: {
-        'Authorization': 'Bearer ' + localStorage.token,
-        'Content-Type': 'application/json'
+const executor = function (baseUrl, deviceId, operation) {
+  const signature = signer(deviceId, operation);
+  return axios({
+    url: baseUrl + "/device/" + deviceId + "/execute",
+    method: "POST",
+    data: signature,
+    skipAuthorization: true,
+    transformRequest: [
+      function (data) {
+        return data;
       },
-      timeout: 10000
-    }
-  )
-}
-var getDoordeckUserByEmail = function (userEmail, visitor) {
-  return axios.post(baseUrl + '/share/invite/' + userEmail + '?visitor=' + visitor, null, {
-    headers: {
-      'Authorization': 'Bearer ' + localStorage.token,
-      'Content-Type': 'application/json'
-    }
-  }).then(response => {
-    return response
-  })
-}
-var getUserByEmail = function (email) {
-  return axios.post(baseUrl + '/directory/query', {'email': email}, {
-    headers: {
-      'Authorization': 'Bearer ' + localStorage.token,
-      'Content-Type': 'application/json'
-    }
-  }).then(response => {
-    return response
-  })
-}
-var getUserById = function (id) {
-  return axios.post(baseUrl + '/directory/query', {'localKey': id}, {
+    ],
     headers: {
-      'Authorization': 'Bearer ' + localStorage.token,
-      'Content-Type': 'application/json'
-    }
-  }).then(response => {
-    return response
-  })
-}
+      Authorization: "Bearer " + localStorage.token,
+      "Content-Type": "application/json",
+    },
+    timeout: 10000,
+  });
+};
+const getUserByEmail = function (baseUrl, userEmail) {
```

Comment:
So, really important point for third-party developers, this API is *not* for them, it will not work, they'll get a forbidden error, instead they need to use the user lookup API below (getUserById).

Reply:
indeed that function wasn't exported.

I've changed the visibility by an underscore to make a better understanding of it:

![image](https://github.com/doordeck/doordeck-sdk-javascript/assets/6842786/2eca6ece-37b5-45ad-bdb1-f1d574652478)

Reply:
`getUserById` *can* be exported, that's the directory lookup function, it's good for everyone and can also be used to query by third party ID, phone number, email or Doordeck ID (but only for user's already in the database, unlike the invite one)

Reply:
there`s `getUser` which does search by email and ID
so basically
`getUser(baseUrl, UUID)`  will get the `getUserById`
Though I can export `getUserById`, just saying in case you didn't see it. Let me know and I'll make the final change

![image](https://github.com/doordeck/doordeck-sdk-javascript/assets/6842786/62d9f8d6-26e3-4af5-bc13-c80dcdf22335)